O365/financials.py

```diff
@@ -0,0 +1,4048 @@
+
+    @unit_of_measure_id.setter
+    def unit_of_measure_id(self, value):
+        self.__unit_of_measure_id = value
+        self._track_changes.add(self._cc('unitOfMeasureId'))
+        self.__unit_of_measure = super()._get_entity(entity_id=self.__unit_of_measure_id, constructor=UnitOfMeasure)
+
+    @property
+    def unit_price(self):
+        return self.__unit_price
+
+    @unit_price.setter
+    def unit_price(self, value):
+        self.__unit_price = value
+        self._track_changes.add(self._cc('unitPrice'))
+
+    @property
+    def quantity(self):
+        return self.__quantity
+
+    @quantity.setter
+    def quantity(self, value):
+        self.__quantity = value
+        self._track_changes.add(self._cc('quantity'))
+
+    @property
+    def discount_amount(self):
+        return self.__discount_amount
+
+    @discount_amount.setter
+    def discount_amount(self, value):
+        self.__discount_amount = value
+        self._track_changes.add(self._cc('discountAmount'))
+
+    @property
+    def discount_percent(self):
+        return self.__discount_percent
+
+    @discount_percent.setter
+    def discount_percent(self, value):
+        self.__discount_percent = value
+        self._track_changes.add(self._cc('discountPercent'))
+
+    @property
+    def discount_applied_before_tax(self):
+        return self.__discount_applied_before_tax
+
+    @property
+    def amount_excluding_tax(self):
+        return self.__amount_excluding_tax
+
+    @property
+    def tax_code(self):
+        return self.__tax_code
+
+    @tax_code.setter
+    def tax_code(self, value):
+        self.__tax_code = value
+        self._track_changes.add(self._cc('taxCode'))
+
+    @property
+    def tax_percent(self):
+        return self.__tax_percent
+
+    @property
+    def total_tax_amount(self):
+        return self.__total_tax_amount
+
+    @property
+    def amount_including_tax(self):
+        return self.__amount_including_tax
+
+    @property
+    def invoice_discount_allocation(self):
+        return self.__invoice_discount_allocation
+
+    @property
+    def net_amount(self):
+        return self.__net_amount
+
+    @property
+    def net_tax_amount(self):
+        return self.__net_tax_amount
+
+    @property
+    def net_amountIncludingTax(self):
+        return self.__net_amountIncludingTax
+
+    @property
+    def shipment_date(self):
+        return self.__shipment_date
+
+    @shipment_date.setter
+    def shipment_date(self, value):
+        self.__shipment_date = parse(value)
+        self._track_changes.add(self._cc('shipmentDate'))
+
+    @property
+    def item(self):
+        return self.__item
+
+    @property
+    def account(self):
+        return self.__account
+
+    @property
+    def unit_of_measure(self):
+        return self.__unit_of_measure
+
+    def update(self):
+        return super()._update()
+
+    def delete(self):
+        return super()._delete()
+
+    def create(self):
+        return super()._create()
```

Comment:
Those are redundant? If they do nothing they can go away

Reply:
I am not sure I got your point. The methods create/update/delete are implemented in the parent class `FinancialsApiComponent` because I did not want to create redundancy by implementing the methods in every sub-class. The methods are hidden (_), because according to the documentation, you cannot CRUD on every object. For entities, that do not allow create/update/delete, the methods were not added.

Reply:
ok ok its create with underscore (_create). nevermind